docs/services/gcp-services-guide.md

```diff
@@ -0,0 +1,536 @@
+```bash
+kubectl cluster-info
+```
+
+### Crossplane
+Using your GKE cluster:
+
+* Install Crossplane from alpha channel using the [Crossplane Installation
+  Guide](../install-crossplane.md#master)
+* Install the GCP stack into Crossplane using the [GCP stack
+  section](../install-crossplane.md#gcp-stack) of the install guide.
+
+We will be using `gcp-infra-dev` namespace to store our cloud-specific resource
+classes, cloud resources and providers. But you can use any namespace name for
+that matter.
+```bash
+kubectl create namespace gcp-infra-dev
+export INFRA_NAMESPACE=gcp-infra-dev
+```
+
+#### GCP Provider
+It is essential to make sure that a GCP service account to be used by 
+Crossplane is configured with all the necessary permissions outlined in the 
+[provider guide](../cloud-providers/gcp/gcp-provider.md).
+
+Using the service account json `crossplane-gcp-provider-key.json`:
+* Generate Base64 encoded value to store in a `Secret`:
+```bash
+export BASE64ENCODED_GCP_PROVIDER_CREDS=$(base64 crossplane-gcp-provider-key.json | tr -d "\n")
+```
+
+* Define a GCP `Provider` and `Secret`:
+```bash
+cat > gcp-provider.yaml <<EOF
+---
+apiVersion: v1
+data:
+  credentials.json: $BASE64ENCODED_GCP_PROVIDER_CREDS
+kind: Secret
+metadata:
+  namespace: $INFRA_NAMESPACE
+  name: gcp-provider-creds
+type: Opaque
+---
+apiVersion: gcp.crossplane.io/v1alpha2
+kind: Provider
+metadata:
+  namespace: $INFRA_NAMESPACE
+  name: gcp-provider
+spec:
+  credentialsSecretRef:
+    name: gcp-provider-creds
+    key: credentials.json
+  projectID: $PROJECT_ID
+EOF
+
+kubectl apply -f gcp-provider.yaml
+
+```
+* Verify GCP provider was successfully registered by the crossplane
+```bash
+kubectl get providers.gcp.crossplane.io -n $INFRA_NAMESPACE
+kubectl get secrets -n $INFRA_NAMESPACE
+```
+
+#### Cloud-Specific Resource Classes
+Cloud-specific resource classes are used to define a reusable configuration for
+a specific managed service. Wordpress requires a MySQL database, which can be
+satisfied by a [Google Cloud SQL Instance](https://cloud.google.com/sql/docs/mysql/) instance.
+
+* Define a GCP CloudSQL class `CloudsqlInstanceClass`:
+```bash
+cat > gcp-mysql-standard.yaml <<EOF
+---
+apiVersion: database.gcp.crossplane.io/v1alpha2
+kind: CloudsqlInstanceClass
+metadata:
+  name: standard-cloudsql
+  namespace: $INFRA_NAMESPACE
+specTemplate:
+  databaseVersion: MYSQL_5_7
+  tier: db-n1-standard-1
+  region: us-central1
+  storageType: PD_SSD
+  storageGB: 10
+  # Note from GCP Docs: Your Cloud SQL instances are not created in your VPC network.
+  # They are created in the service producer network (a VPC network internal to Google) that is then connected (peered) to your VPC network.
+  privateNetwork: projects/$PROJECT_ID/global/networks/$NETWORK_NAME
+  providerRef:
+    name: gcp-provider
+    namespace: $INFRA_NAMESPACE
+  reclaimPolicy: Delete
+EOF
+
+kubectl apply -f gcp-mysql-standard.yaml
```

Comment:
Are these yamls also in the repo in some examples folder somewhere? It may be worth telling people something like "for more details, see the full example at . . ."

Reply:
No, these are extracted from Wordpress workload example YAMLs. We'd need to consider whether it's worth creating specific folder for this guide.